src/sumologic-span-processor/index.spec.ts

```diff
@@ -192,6 +195,33 @@ describe('SumoLogicSpanProcessor', () => {
       document.dispatchEvent(new Event('visibilitychange'));
       span.end(endTime); // end time is before page was hidden
       spanProcessor.onEnd(span);
+      jest.runAllTimers();
+
+      // Although page was hidden after span end time, we still marked it as 'hidden'.
+      // It's a special case that only applies for root spans.
+      // It was introduced to support `documentLoad` page visibility.
+      // Because `documentLoad` span ends when page was loaded, there are may be gaps in spans, so there would be no span
+      // where we may put the 'pagehide' event.
+      expect(span.attributes['document.visibilityState']).toBe('hidden');
+    });
+
+    test('for span started with custom time', () => {
+      setSystemTime('2022-01-01 10:00');
+      setDocumentVisibilityState('hidden'); // page is initially hidden
+      document.dispatchEvent(new Event('visibilitychange'));
+
+      setSystemTime('2022-01-01 10:01');
+      const startTime = hrTime(); // our root span will start at this point
+
+      setSystemTime('2022-01-01 10:02');
+      setDocumentVisibilityState('visible'); // page is visible when span is creating
```

Comment:
Can you explain why this is expected?

Reply:
In this test we make sure that the given span start time is considered and not the `spanProcessor.onStart` invocation time. So in practice, span here is created when page was hidden.

Reply:
> So in practice, span here is created when page was hidden

Why does the comment say:
> page is visible when span is creating

then?